### DISCUSSION TITLE
ChatGPT Ideas Collection

### DISCUSSION THREAD

**Discussion by lencx** [lencx/ChatGPT#37] [Ideas]
- ChatGPT model - [Awesome ChatGPT Prompts](https://github.com/f/awesome-chatgpt-prompts)

**Comment by f** [comment]
The awesome-chatgpt-prompts repo includes all the prompts as csv file in it. You can simply import from the repo and allow your users to use them quickly. The slash command can be generated from the title I guess.

**Reply by lencx** [reply]
We can generate a short command from the title and then the user can also edit it according to their own habits. I can currently use `prompts.csv` for the initial functionality, but it would be nice if the csv file could include a more standardized cmd short command later.

**Comment by PianoMastR64** [comment]
Name image capture files the same name as the chat. Or at least include it in the file name.

**Reply by lencx** [reply]
It may be better if custom file name generation rules are supported.

**Reply by leilei175** [reply]
the chat's name would be great . It seems chatgpt would summarize the question and response. It would be great to save the image or pdf as the name displayed on the left side.

**Comment by MaCoVibes** [comment]
## Auto-Save type function.

Due to the recent, and supposedly temporary removal of history, I would love to see history, per chat, synchronized/saved in unique folders named after the chat name on the tab, and all dialog accessible offline like an auto-save. Then, if ChatGPT is down like it was earlier, I could go into that local chat conversation folder **on my drive**, and retrieve any portion of that prompt/response chain. Date/Time stamping would be helpful, as well as the 1/3, 2/3, 3/3 versions of a prompt.

Losing a thread to your history is like losing everything to a neurodivergent. We have thousands of waffle-pockets interconnected by these threads. If we lose the thread, we might as well have lost the storage locker because we'll keep trying to use a key on doors that can't be opened, and all be we have a shimmer of a memory that the key opens "something", and then it become an obstacle in front of any other form of productivity we have. I have 1000 posted notes with information on them. I may not need them for days/months/years, but I know exactly where to go when I do. If the note pile is missing, it can take me weeks/months to recover. 

Having said the last part, maybe an additional way to save history in this manner could be in the cloud and as suggested above. This would be a feature I would pay for even though this project is free.

**Reply by lencx** [reply]
I am currently implementing this feature to export data to markdown with secondary editing and preview support. I'm building a built-in markdown editor to turn it into a note taking software. https://github.com/lencx/ChatGPT/issues/187

**Reply by MaCoVibes** [reply]
@lencx Thank you for the addition of this. I am lacking skill to identify where I can make a change in the script for a path that is not on my local machine. I have the install on my external drive, so I assumed that it would save the dialog on the external drive. When I view the config, it seems that that the save path is local to C:\. How might I edit, if it is possible, to another path for saving history without breaking anything else in the app or existing configuration where I do not wish to make changes?

**Reply by lencx** [reply]
Path changes can be complicated and a lot of core code needs to be handled. Also, tauri does not recommend that applications read or write arbitrary paths to the system, as it is not secure.

**Reply by MaCoVibes** [reply]
@lencx Thank you for such a quick reply. I will then consider creating the directory on the external drive, and then find a way to syncrhonize the data from the local to the external directory.

**Reply by lencx** [reply]
I save all configurations locally in json file format just to accommodate data export. I will consider your suggestion to add a custom root directory to meet some special needs of users. But it may take a while before it can be scheduled, as I am still developing more important features for it.

**Reply by MaCoVibes** [reply]
@lencx The fact that you were able to accommodate my #187 suggestion is worth its weight in gold. Stay on track with other priorities, as you have already answered my question, and I have a temporary solution already about to be deployed. Because I am currently using a company owned device, it is nice to consider an alternative location to save history. Even if you are unable to perform this consideration in the next 6 months, a year or ever, this is okay with me because you have a great product already. Stay tuned for incoming coffee 🥇  ☕ ☕ ☕

**Reply by MaCoVibes** [reply]
@lencx Of course, I have an additional question. Might you know why it is that when I click to` Export Markdown` that Windows opens a File Explorer window? It is not asking me where to save the file, but it always seems to do this.

- Type in prompt.
- GPT responds to prompt.
- Click `Export Markdown` and then the window opens and shows ThisPC as if I opened File Explorer and focus on ThisPC in the tree on the left side.

![image](https://user-images.githubusercontent.com/122420403/213532372-adba3847-8d21-4bf9-a180-d5fdebf8ed42.png)

**Reply by lencx** [reply]
It is opening the system path of the exported file on mac, but it seems to have problems on windows, I can optimize it in the next version.

**Comment by rustyorb** [comment]
It was listed that exporting the conversation was a feature, or did I read that incorrectly? If it is a feature, how do I export the conversation? Aside from copy, paste I guess. Thanks for your worrk.

**Reply by lencx** [reply]
Because the page layout was updated on the official website last night, the export button was not working, the next version will fix it.

**Comment by PapuChad22** [comment]
You should add a feature in which you can search for conversations related for something. For example, I have like 40 different chatgpt conversations so it's a mess. If I want to search the conversation in which i talked about aliens, for example, i search in the search tab aliens, and it shows me conversations about that. Kinda works like CTRL + G while using google.

Also another feature could be creating our own '/' commands for chatgpt

**Comment by bolohori** [comment]
Hello @lencx ! 

Some suggestions for improving the app if you're' still taking them ;)  
#### Exporting chat to md/pdf/image improvements:

- Naming the output export file yourself and choosing a location on where to save it  - current naming tends to make it a bit confusing
- Ability to cherry-pick pairs of prompt-output out of a conversation when saving it - sometimes you get stuck with some irrelevant/unwanted/repeated answers that only make for garbage in the end
- Maybe as a workaround, ability to delete (or just hide) irrelevant outputs

#### These might not worth doing, but here they are:

**The Side panel** - similar functionality as Microsoft Edge's collection (add page, pin content, add note)
- Pin chosen prompt-output into the side panel ( or put it on top of the chat maybe ? )
- As a workaround, maybe organize chosen outputs in a slider like manner as you made the edit prompt/2nd answer 
- A separate side panel for taking your own in-app markdown notes based on current chat 
- 
**Tricking chatGPT into reading a very long text file at once**.  - based on "I'm sorry, but as an AI language model, I do not have the capability to access external links or perform actions outside of this text-based chat interface."

**Reply by DiverBW** [reply]
I very much agree with this suggestion of being able to name the PDF file that you save a chat to, and the directory.  Or even if we stay with the cryptic filename, just adding a settings option that lets us configure the directory that is used would help. I'd much prefer that the directory default to the standard "Downloads" directory for the user rather than under the hidden .chatgpt folder.

I am relatively new to the Mac world. I presume that there is some trick to getting Finder to navigate into folders that start with a "." to hide them, but I've not googled for that yet. My bad! :)

**Comment by jonathanplesel** [comment]
The possibility of adding custom CSS to remove and modify elements.

<img width="441" alt="Capture d’écran 2023-03-05 à 12 59 00" src="https://user-images.githubusercontent.com/94924350/222959092-8673e413-afe4-401d-ba42-cbe9dd77e7fd.png">

**Comment by mightymeyti** [comment]
please add an option so we can use proxy servers like socks5

**Reply by sebastiendurand78** [reply]
https://github.com/lencx/ChatGPT/discussions/705

**Comment by Jewseph** [comment]
I love the fact you use text-to-speech in this. My idea was it would be great if you could speed up the speech and have it talk as it is typing out the responses!
Thanks,
Joe

**Reply by ridwan47** [reply]
The 1x TTS speed is driving me insane!

**Comment by PianoMastR64** [comment]
Make it so I can change the size of various elements. The chat history is too narrow for example. And sometimes I wish I could widen the chat itself

**Comment by SummerRayMorrow** [comment]
I have been using your app and truly appreciate the work you have put into it. I have taken the time to compile a list of suggestions that I believe could further improve the app and enhance the user experience. Please find the suggestions grouped by topic below. As we may not share a common language, I have tried to phrase them in a way that is translation app friendly to ensure clear communication. I hope you find these suggestions helpful and consider implementing them in future updates. Thank you for your dedication and continued efforts in making this app great.

Here are the suggestions grouped by topic:

User Interface & Customization:
1. [ ] Include a button in the top navigation bar's drop-down menu for toggling the side navigation bar's visibility.
2. [ ] Enable horizontal resizing of the side navigation bar for better customization.
3. [ ] Allow vertical resizing of the main chat's text box for improved customization.
4. [ ] Offer options for users to personalize font size, background colors, text colors, and other UI elements.
5. [ ] Offer an even darker mode option for users who prefer darker color schemes.

Organizing Conversations:
1. [ ] Add functionality for users to create folders and automatically sort conversations for better organization.
2. [ ] Implement a search feature for finding specific conversations or folders based on conversation names or content.
3. [ ] Include multi-selection checkboxes for managing conversations, such as moving or deleting them.
4. [ ] Allow users to archive conversations, keeping them accessible but out of the main view.
5. [ ] Include options to bookmark or pin specific messages or conversations for quick access, and provide sorting options.
6. [ ] Add a confirmation step before deleting certain conversations and folders to prevent accidental deletion.

Interaction & Messaging:
1. [ ] Address the inconsistency in the taskbar app's behavior between large and small screen modes regarding the "Enter" key and "Tab"+"Enter" key combination. Improve the messaging to clarify the use of the "Tab" key for sending messages in small mode.
2. [ ] Add emoji or reaction support in conversations.
3. [ ] Implement a feature for sharing messages or entire conversations with other conversations.
4. [ ] Allow users to edit or delete messages within a specific time frame that doesn't count towards the message cap.

Notes & Files:
1. [ ] Allow the chatbot to access files in the "notes" section.
2. [ ] Tray Pop-up & Taskbar:
3. [ ] Enable users to resize the tray pop-up by dragging its edges, and optionally undock it from the tray.
4. [ ] Provide an option to keep the tray icon visible even when the taskbar icon is hidden.

Notifications & Feedback:
1. [ ] Add a notification system that alerts users when the chat finishes printing.
2. [ ] Improve the "Provide additional feedback" feature's messaging to clarify its purpose, whether the feedback is for the AI or developers, and the confidentiality of the feedback.
3. [ ] Enhance the "Provide additional feedback" feature by clarifying the benefits of using thumbs up and down buttons for user feedback

Prompts & Responses:
1. [ ] Implement a feature to automatically continue truncated responses due to length limitations.
2. [ ] Offer a queue for prompts instead of warning "one message at a time."
3. [ ] Add a feature to automatically split and queue prompts that exceed 2000 characters, ensuring users can still submit longer prompts without issues.

Offline Use:
1. [ ] Enable users to store a model locally for offline use.

**Reply by mofitn3577** [reply]
All great Ideas. Most are things that should have been added by now idk what you have to do to get a working program. I'm about to hire a developer to start building for me though because it's very limiting when you don't have the programming skills to just go build something to accomplish what you need to do in a few minutes with python and the openAI API. 

Is anyone still working on adding features to this or is it basically deal with it or go find something else at this point? So much potential and nothing really being done seemingly. Everyone is apparently using something else and not bothering with this now I feel like the only one still using it trying to optimize it to make it usable for my purpose. 

The big one is "Allow the chatbot to access files in the "notes" section." Shit allow it to access something that we can put plain text or documents or something. It drives me insane having to repeat everything all over again constantly.


The absolute worst is when I'm working on something and have a good flow going and the chatbot actually has a perfect understanding of what I'm working on and the end goal and I'm actually productive for once and then I realize I need it to do something that isn't accomplished by one of the 3 plugins at a time that can be active. Then I have to open a new chat or go to one that I remembered to label that has the plugin that I need open and try to get it to do what just took an hour of prompting to accomplish in the other chat. Plus I lose the chat that has the prompt I need so I have to click on a ton to find it or start a new one and go through the whole list of plugins to find it and start from scratch with prompting. And its like playing the lottery which type of chatbot you'll get even though its the same exact model so going to a new chat when you have one dialed in that's not going off on random tangents or forgetting what you just told it 2 minutes ago really sucks. 

And the plugin thing I really don't understand. Did someone just arbitrarily decide that only 3 plugins should be allowed to be used on a chat and then decide to make it impossible to change them without starting from scratch with a new chat? Would it really be impossible to just activate a different plugin as the need came up in a chat and that's why this was done or could it actually just easily allow you to call on any of the installed plugins as needed without switching to another chat? Seems like it should be able to but what do i know. If i knew anything i would be able to code something to do what I need instead of trying to make this work I guess haha It really is crazy when you think about the 3 plugin thing though. 

I would be fine with it if you could change them as needed on the fly but if you think about it the way it is now would be like chrome saying you can only have 3 of your extensions running at a time so have fun trying to get anything done with 50 different web browsers open bouncing back and forth and trying to remember which one has the extensions running that you need to use at that particular moment. Or actually more accurately it's like saying you can only have one chrome instance open at a time and only 3 extensions active that you have to select in advanced and when you need a different one you have to close it entirely and open another instance and start from scratch without your tabs or anything else reopening and all the cookies and autofill info wiped completely so you're starting from scratch entirely and everything you were just working on starts all over. It's crazy.

**Comment by gmy77** [comment]
Like in the WEB a series of "personalized instructions" adding all possible about the user... many numbers of information's added. best is. I find this "fantastic, at least..." on the web. Any new answer generation about my question of anything, contain, correctly, personalized logical based only on my personal character, based on my way of think about any things... i noted this function IF feeder well... AMAZING... AMAZING FOR SURE. I talk to ChatGPT... like to talk to any normal person in this world... There was a moment... from myself... I said to myself that "it was a truly incredible experience"... of course. I must admit, I entered a lot of data about myself in the ChatGPT personalization feature, logical, personal and many many more... on the web and I think this has generated something incredibly... crazy.

**Comment by Lenoxiduds** [comment]
Hi. First of all, thank you for this software, it's really great to have ChatGPT as an application.
I just wanted to know if it was possible to minimize the window to the system tray, rather than the taskbar?
